cosmo_tester/test_suites/test_security/ad_auth_test.py

```diff
@@ -0,0 +1,152 @@
+        self.localenv.execute('install',
+                              task_retries=9,
+                              task_retry_interval=9)
+
+        return self.localenv.outputs().get('ldap_endpoint')
+
+    def _terminate_active_directory_service(self):
+        self.localenv.execute('uninstall',
+                              task_retries=9,
+                              task_retry_interval=9)
+
+    def _get_active_directory_inputs(self):
+        return {
+            'instance_type': self.env.medium_instance_type,
+            'image_id': self.env.windows_server_2012_r2_image_id,
+            constants.AWS_CONFIG_PROPERTY:
+                self._get_aws_config()
+        }
+
+    def _get_aws_config(self):
+
+        region = get_region(self.env.ec2_region_name)
+        return {
+            'aws_access_key_id': self.env.aws_access_key_id,
+            'aws_secret_access_key': self.env.aws_secret_access_key,
+            'ec2_region_name': region
+        }
+
+    ######################################
+    # override default security settings
+    ######################################
+
+    # Use the LDAP authentication plugin to authenticate against the running
+    # active directory instance.
+    def get_authentication_providers(self):
+        return [
+            {
+                'implementation': 'authentication.ldap_authentication_'
+                                  'provider:LDAPAuthenticationProvider',
+                'name': 'ldap_authentication_provider',
+                'properties': {
+                    'directory_url': '{0}'.format(self.ad_ip)
+                }
+            }
+        ]
+
+    # The LDAP authentication plugin is installed via wagon since it's
+    # dependencies require gcc and python-dev.
+    def get_rest_plugins(self):
+        return {
+            'ldap_authentication_provider':
+                {
+                    'source': 'http://adaml-bucket.s3.amazonaws.com/'
+                              'cloudify_ldap_plugin-1.3-py27-none-linux_'
+                              'x86_64-centos-Core.wgn',
+                    'install_args': 'cloudify-ldap-plugin --use-wheel'
+                                    ' --no-index --find-links=wheels/ --pre'
+                }
+            }
+
+    # Use file based userstore
+    def get_userstore_driver(self):
+        return {
+            'implementation': 'flask_securest.userstores.file_userstore:'
+                              'FileUserstore',
+            'properties': {
+                'target_path': '/opt/manager/userstore.yaml'
+            }
+        }
+
+    # The ldap authentication plugin requires openldap-devel.
+    def get_userdata(self):
+        return '#!/bin/bash sudo yum install openldap-devel -y'
+
+    # this test does not require the passwords be stored in the userstore
+    def get_userstore_users(self):
+        return [
+            {
+                'username': auth_test_base.ADMIN_USERNAME,
+                'groups': [
```

Comment:
these groups don't exist in your active directory, so they should not be part of the userstore